### PR TITLE
Add support for AlarmWidget in awsx.cloudwatch

### PR DESCRIPTION
**Pull request by catmeme**

### CODE REVIEW

File: nodejs/awsx/cloudwatch/widgets_json.ts [hunk 1]
```diff
@@ -21,14 +21,19 @@ import { YAxis } from "./widgets_graph";
...
     x: pulumi.Input<number>;
     y: pulumi.Input<number>;
     width: pulumi.Input<number>;
     height: pulumi.Input<number>;
     properties: Record<string, any>;
 }
 
+export interface AlarmWidgetJson extends WidgetJson {
+  type: pulumi.Input<"alarm">;
+  properties: pulumi.Input<{ alarms: pulumi.Input<string>[] }>;
```
Comment (leezen) [inline nodejs/awsx/cloudwatch/widgets_json.ts:34]: Looking at https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/CloudWatch-Dashboard-Body-Structure.html#CloudWatch-Dashboard-Properties-Alarm-Widget-Object -- it seems there are a number of other keys that can actually exist under `properties`. Do you mind extending this to be similar to `MetricWidgetJson` where you defined an `AlarmWidgetPropertiesJson` and include those additional fields (as optional)?

**Review by leezen — COMMENTED** [review]
Thanks for contributing this! I had a small question about the actual schema for the `AlarmWidgetJson` type.

File: nodejs/awsx/cloudwatch/widgets_json.ts (hunk 1 above)
Comment (catmeme) [reply nodejs/awsx/cloudwatch/widgets_json.ts:34]: I've pushed an update, wasn't sure if I should be breaking out the values of `states` and `sortBy` further.  Also, should I squash my commits?

Please advise.

**Comment by leezen** [comment]
When I merge this, I'll do so via "Squash and merge" so no need to worry about squashing your commits.

File: nodejs/awsx/cloudwatch/widgets_json.ts (hunk 1 above)
Comment (leezen) [reply nodejs/awsx/cloudwatch/widgets_json.ts]: > I've pushed an update, wasn't sure if I should be breaking out the values of states and sortBy further.

Do you mean into their own types? I think it's fine to inline as you have it for now.

**Review by leezen — APPROVED** [review]

**Comment by leezen** [comment]
This LGTM, so happy to merge if you are.

File: nodejs/awsx/cloudwatch/widgets_json.ts (hunk 1 above)
Comment (catmeme) [reply nodejs/awsx/cloudwatch/widgets_json.ts]: Yes, that's what I meant, they could be enums.

**Comment by catmeme** [comment]
Go ahead an merge, thank you!

**Comment by leezen** [comment]
@catmeme Thanks for contributing this!

**Pull request merged**